packages/core/src/shared/sagemaker/client/sagemaker.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import { fromContainerMetadata } from '@aws-sdk/credential-provider-imds'
+import { Service } from 'aws-sdk'
+import globals from '../../extensionGlobals'
+import SageMakerClient, { DescribeDomainRequest, DescribeDomainResponse } from './sagemakerclient'
+import apiConfig = require('./service-2.json')
+import { ServiceOptions } from '../../awsClientBuilder'
+
+export interface SageMakerCookie {
+    authMode?: 'Sso' | 'Iam'
+    expiryTime?: number
+    ssoExpiryTimestamp?: number
+    studioUserProfileName?: string
+    redirectURL?: string
+    AccessToken?: string
+    StudioSessionToken?: string
+}
+
+// Manually expire the token every 5 minutes as token refresh is handled externally by Sagemaker
+const TOKEN_EXPIRY = 5 * 60 * 1000
```

Comment:
Use constants in https://github.com/aws/aws-toolkit-vscode/blob/e6c3101e8950acea5158407b7099d66a26663a8c/packages/core/src/shared/datetime.ts#L8

e.g.
```suggestion
const TOKEN_EXPIRY = oneMinute * 5
```